### ISSUE TITLE
AnnotationOnSeparateLine should respect comments

### ISSUE THREAD

**Issue by arturbosch** [pinterest/ktlint#498]
Following file has three annotations with a comment next to it: 
https://github.com/arturbosch/detekt/blob/8ae133c2876a94814973ba397c6809cd4a5e514b/detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/ThresholdedCodeSmellAssert.kt#L8-L9

Unfortunately KtLint does not respect this.

The first testcase in AnnotationRuleTest modified to get a violation reported:
```
    @Test
    fun `lint single annotation may be placed on line before annotated construct`() {
        assertThat(
            AnnotationRule().lint(
                """
                @Suppress("UnsafeCast") // False positive, see issue #1137 
                class A {
                    @JvmField
                    var x: String
                }
                """.trimIndent()
            )
        ).isEmpty()
    }
```

The check which fails for comments:
https://github.com/pinterest/ktlint/blob/a726c87514de5a9ef547529bb4bff299df54dd4d/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/AnnotationRule.kt#L50-L52
